pkg/ccl/changefeedccl/kvfeed/kv_feed_test.go

```diff
@@ -645,3 +700,151 @@ func TestCopyFromSourceToDestUntilTableEvent(t *testing.T) {
 		})
 	}
 }
+
+func TestFrontierQuantization(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	mockQuantization := func(ts hlc.Timestamp, quantize time.Duration) hlc.Timestamp {
+		ts.Logical = 0
+		ts.WallTime -= ts.WallTime % int64(quantize)
+		return ts
+	}
+
+	makeCheckpointEvent := func(key []byte, endKey []byte, ts int) *kvpb.RangeFeedEvent {
+		return &kvpb.RangeFeedEvent{
+			Checkpoint: &kvpb.RangeFeedCheckpoint{
+				Span:       roachpb.Span{Key: key, EndKey: endKey},
+				ResolvedTS: hlc.Timestamp{WallTime: int64(ts)},
```

Comment:
maybe add a logical time to this so you can test that it gets zero'd

Reply:
Done.